custom_components/hyxi_cloud/sensor.py

```
@@ -1459,15 +1459,29 @@ def _update_native_value(self):
         key = self.entity_description.key
         value = metrics.get(key)
 
-        # 🚀 Fallback Logic for Micro Inverters (acE -> efpv)
-        # If acE is not provided or zero, attempt fallback to efpv for Micro Inverters.
+        # Fallback Logic for Micro Inverters (acE -> efpv, gridF -> f)
         if key == "acE" and (value is None or str(value) == "0.0"):
             if getattr(self, "_device_type", None) in (
                 "grid_connected_inverter",
                 "micro_inverter",
             ):
```

Ruff (0.15.20)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)